### DISCUSSION TITLE
Show timer as icon badge - if any timed page open

### DISCUSSION THREAD

**Discussion by woj-tek** [proginosko/LeechBlockNG#369] [Ideas]
Hi,
There is an option _Show timer as icon badge (if time left is less than 10 minutes)_ which is great but there is also _Count time spent on these sites only when browser tab is active_ and in case if the second one is disabled (i.e. page is unfocused and the timer is still being counted) there is no indication on the extension icon.

It would be also nice if there could be an option to include that timer without 10 minutes limit (or configure the limit)

EDIT: possible solution: in `background.js` in `updateTimer()` method calculate also `globalSecsLeft` and if main `secsLeft` doesn't exist or `globalSecsLeft` is lower then display `globalSecsLeft`. This would possibly impose performance penalty due to iterating over all tabs in `gTabs` array.

**Comment by proginosko** [comment]
The reason for the 10 minutes is that the icon badge can only accommodate 4 characters (hence "M:SS").

I don't understand your first suggestion. Are you asking for the time left to be shown for non-active tabs? I don't see how that would work when there are _multiple_ tabs with sites in block sets. Which time-left value would LB display?

**Comment by woj-tek** [comment]
> The reason for the 10 minutes is that the icon badge can only accommodate 4 characters (hence "M:SS").

Fair enough :-)

> I don't understand your first suggestion. Are you asking for the time left to be shown for non-active tabs? I don't see how that would work when there are multiple tabs with sites in block sets. Which time-left value would LB display?

As I mentioned - the one with the lowest limit.

Maybe I'll shed a bit of light - I have a limit for a couple of websites and sometimes I open a webpage from blocked domain in the background and it times-out the limit I had without me knowing it. It's result of disabling "process only active tabs" because - I figured a simple way to circumvent the timer - simply make browser loose the focus and I can still read the content without the limit. Bottomline - solving #49 would most likely render this issue (badge timer) invalid.

**Comment by woj-tek** [comment]
Alternatively - instead of showing timer itself do following:
* if there is a timer on current page - show it
* if there is no timer on current page but active for (any) inactive page - show a dot/symbol indicating that there is timer active somewhere.
